WordPress/gutenberg
# Prevent Pre-Publish Panel from Displaying Incorrect Information After Navigating away

Fixes: #64913 


## What?
This PR addresses a bug highlighted in #64913 where the pre-publish panel displays incorrect or broken information when switching to edit a template part. Specifically, if you open the pre-publish checks and immediately switch to edit a template part, the panel continues to show details about the template part, instead of closing or updating appropriately.

## Why?
This fix improves the user experience by preventing confusion caused by incorrect information in the pre-publish panel. It ensures the behavior aligns with expectations, where the pre-publish is closed when navigating away.

## How?
The issue is resolved by ensuring that when the user navigates away from editing the page, the pre-publish panel is closed. This change ensures that the publish flow is not interrupted and the pre-publish checks are only shown for the current page or post, not the template part.

## Testing Instructions

1. Open a draft page in the editor
2. Ensure pre-publish checks are enabled
3. Click on `publish` to open the pre-publish panel
4. Select the header then click 'Edit' in the block toolbar
5. Now the pre-publish panel closes when navigating away

## Screenshots or screencast 

https://github.com/user-attachments/assets/f750d1c9-9138-474a-87f3-696e5d2801dd
